SparkBuilderGeneratorPlugin/test/com/helospark/spark/builder/handlers/it/BuilderWithSuperSetterE2ETest.java

```diff
@@ -0,0 +1,202 @@
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.dom.TypeDeclaration;
+import org.mockito.Mock;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import com.helospark.spark.builder.handlers.GenerateRegularBuilderHandler;
+import com.helospark.spark.builder.handlers.GenerateStagedBuilderHandler;
+import com.helospark.spark.builder.handlers.it.dummyService.NoDialogOperationPerformedStagedBuilderDialogAnswerProvider;
+import com.helospark.spark.builder.handlers.it.dummyService.TypeExtractorAnswerProvider;
+
+public class BuilderWithSuperSetterE2ETest extends BaseBuilderGeneratorIT {
+    private NoDialogOperationPerformedStagedBuilderDialogAnswerProvider dialogAnswerProvider = new NoDialogOperationPerformedStagedBuilderDialogAnswerProvider();
+    @Mock
+    private IType firstSuperClassType;
+    @Mock
+    private IType secondSuperClassType;
+    @Mock
+    private ICompilationUnit firstSuperClassICompilationUnit;
+    @Mock
+    private ICompilationUnit secondSuperClassICompilationUnit;
+
+    private TypeExtractorAnswerProvider typeExtractorAnswerProvider;
+
+    @BeforeMethod
+    public void beforeMethod() throws JavaModelException {
+        super.init();
+        underTest = new GenerateRegularBuilderHandler();
+
+        // first super class
+        given(firstSuperClassType.getCompilationUnit()).willReturn(firstSuperClassICompilationUnit);
+        given(firstSuperClassType.getElementName()).willReturn("TestSuperClass");
+
+        // second super class
+        given(secondSuperClassType.getCompilationUnit()).willReturn(secondSuperClassICompilationUnit);
+        given(secondSuperClassType.getElementName()).willReturn("TestSuperSuperClass");
+
+        // When getting the superClass return the correct type
+        Map<String, IType> map = new HashMap<>();
+        map.put("TestClass", firstSuperClassType);
+        map.put("TestSuperClass", secondSuperClassType);
+        typeExtractorAnswerProvider = new TypeExtractorAnswerProvider(map);
+
+        given(iTypeExtractor.extract(any(TypeDeclaration.class)))
+                .willAnswer(a -> typeExtractorAnswerProvider.provideAnswer(a));
+
+        given(preferenceStore.getBoolean("org.helospark.builder.includeSetterFieldsFromSuperclass")).willReturn(true);
+    }
+
+    @Test(dataProvider = "superClassWithSetterDataProvider")
+    public void testSuperClassWithSetters(String inputFile, String superClassFile, String expectedOutputFile) throws Exception {
+        // GIVEN
+
+        String superClassInput = readClasspathFile(superClassFile);
+        super.setCompilationUnitInput(firstSuperClassICompilationUnit, superClassInput);
+
+        String input = readClasspathFile(inputFile);
```

Comment:
I think an object could be better.

-String inputfileName
-String superClassFileName
-String expectedOutputFileName
+inputfile();
+expectedOutputFile()
+superClassFile()

Reply:
`readClasspathFile` and `setCompilationUnitInput` are coming from the abstract base test class :innocent: 
Therefore not available in the dataprovider.

Reply:
Composition over inheritance?

Reply:
:+1: 
If no other objections I'll merge it and start to roll out the new version.
If testing improvement is required it will be done as a separate techdebt.